indico/modules/rb_new/client/js/components/RoomBasicDetails.jsx

```diff
@@ -67,26 +69,75 @@ AnnotatedIcon.propTypes = {
 
 export default class RoomBasicDetails extends React.PureComponent {
     static propTypes = {
-        room: PropTypes.object.isRequired
+        room: PropTypes.object.isRequired,
+    };
+
+    state = {
+        imageModalOpen: false,
+        imageLoaded: false,
     };
 
+    renderImage() {
+        const {room} = this.props;
+        const {imageModalOpen, imageLoaded} = this.state;
+        const isClickable = room.hasPhoto;
+        const spriteImage = (
+            <SpriteImage styleName="clickable"
+                         pos={room.spritePosition}
+                         height="100%"
+                         width="100%"
+                         fillVertical
+                         onClick={() => {
+                             if (isClickable) {
```

Comment:
What about returning earlier and saving the `else` part?
```jsx
if (!isClickable) {
    return null;
}
this.setState({imageModalOpen: true});
```

Reply:
I forgot to follow the earlier suggestion that was this: 
onClick={isClickable ? (() => this.setState({openImageModal: true})) : undefined}
Thanks for pointing it out and reminding me about it anyway!

Reply:
True, even better